core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -750,6 +755,9 @@ void CudaInternal::finalize() {
       m_team_scratch_current_size[i] = 0;
       m_team_scratch_ptr[i]          = nullptr;
     }
+
+    cudaFree(m_scratch_locks);
```

Comment:
Why is this one not wrapped into a `KOKKOS_IMPL_CUDA_SAFE_CALL`

Reply:
Good point, will fix.